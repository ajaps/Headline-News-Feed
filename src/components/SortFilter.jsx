import React from 'react';
import PropTypes from 'prop-types';
import * as myActions from '../actions/Headlines';

import ArticleStore from '../stores/Articles';

/**
 * Represents the Available sort filter
 */
export default class SortFilter extends React.Component {
  constructor() {
    super();
    this.bySort = this.bySort.bind(this);
  }
/**
 * sends an action to change the preffered sort filter based on user selection
 * @param {object} e The selected sort Filter
 * @return {void}
 */
  bySort(e) {
    const sortBy = e.target.dataset.sort;
    const SourceId = ArticleStore.getSourceId();
    myActions.setApiSortBy(SourceId, sortBy);
  }

  render() {
    const articleSortFilters = this.props.sortFilter;
    let allAvailableSortFilter;

    if (articleSortFilters) {
      allAvailableSortFilter = articleSortFilters.map(eachSortItem =>
        <li className={eachSortItem} key={eachSortItem}>
          <a data-sort={eachSortItem} onClick={this.bySort} ref={eachSortItem}>
            { eachSortItem } </a>
        </li>
      );
    }

    return (
      <div className="container">
          <div className="dropdown pull-right sortByFeature">
            <a aria-expanded="false" aria-haspopup="true"
                className="dropdown-toggle" data-toggle="dropdown" role="button"
            > Sort Articles By
            <span className="caret" /></a>
            <ul className="dropdown-menu">
              {allAvailableSortFilter}
            </ul>
          </div>
      </div>
    );
  }
}

SortFilter.propTypes = {
  sortFilter: PropTypes.array,
};
