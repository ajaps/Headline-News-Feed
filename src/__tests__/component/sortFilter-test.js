import { mount, shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import SortFilter from '../../component/SortFilter.jsx';

describe('Category', () => {
  let app;
  const filter = ['top', 'latest'];

  beforeEach(() => {
      app = shallow(<SortFilter sortFilter={filter} />);
  });

  it('should render as expected', () => {
    app = shallow(<SortFilter sortFilter={filter} />);
    const tree = toJson(app);
    expect(tree).toMatchSnapshot();
  });

  it('should have extacly three(plus one) dropdown children element', () => {
    expect(app.find('li').length).toEqual(4);
  });
});
