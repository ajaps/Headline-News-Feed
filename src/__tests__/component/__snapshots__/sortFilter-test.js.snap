// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category should render as expected 1`] = `
<div
  className=" container navbar2"
>
  <div
    className="masthead"
  >
    <li
      className="dropdown sortByFeature"
    >
      <a
        aria-expanded="true"
        aria-haspopup="true"
        className="dropdown-toggle"
        data-toggle="dropdown"
        role="button"
      >
         Sort Articles By 
        <span
          className="caret"
        />
      </a>
      <ul
        className="dropdown-menu"
      >
        <li
          className=""
        >
          <a
            onClick={[Function]}
          >
            Top
          </a>
        </li>
        <li
          className=""
        >
          <a
            onClick={[Function]}
          >
            Latest
          </a>
        </li>
        <li
          className="disabled disableClick"
        >
          <a
            onClick={[Function]}
          >
            Popular
          </a>
        </li>
      </ul>
    </li>
  </div>
</div>
`;
